code-experiments/test/unit-test/test_coco_utilities.c

```diff
@@ -444,11 +444,30 @@ static void test_coco_is_orthogonal(void **state) {
   double *M;
 
   M = coco_allocate_vector(4);
+
+  /* Case of an orthogonal diagonal matrix (identity matrix) */
   M[0] = 1.;
   M[1] = 0.;
   M[2] = 0.;
   M[3] = 1.;
   assert(coco_is_orthogonal(M, 2, 2));
+
+  /* Case of a non-orthogonal square matrix */
+  M[0] = 1.;
+  M[1] = 1.;
```

Comment:
why not `0.01`? Tests are supposed to stress the code, not to make it easy to pass.